src/plone/api/user.py

```diff
@@ -232,7 +232,7 @@ def get_roles(username=None, user=None, obj=None, inherit=True):
         user = portal_membership.getMemberById(username)
 
     if user is None:
-        raise UserNotFoundError
+        raise UserNotFoundError(username or user and user.getUserName())
```

Comment:
It does not make sense to try and access `user` within a condition that already ensures, that `user is None`. Also, this does not handle the edge case, where both `user` and `username` are `None` - though logging `None` may be valid.